presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -227,6 +236,17 @@ public TaskStatus process()
                     return TaskStatus.finished();
                 }
                 ListenableFuture<?> future;
+                /*
+                 * Block until one of below conditions is met:
+                 * Arrival of first non "all" filter
+                 * Completion of DynamicFilter
+                 * Timeout after waiting for the configured time
+                 */
+                long timeLeft = dynamicFilteringProbeBlockingTimeoutMillis - stopwatch.elapsed(MILLISECONDS);
+                if (timeLeft > 0 && !dynamicFilter.isComplete() && dynamicFilter.getCurrentPredicate().isAll()) {
+                    future = toListenableFuture(dynamicFilter.isBlocked().orTimeout(timeLeft, MILLISECONDS));
```

Comment:
does `orTimeout(timeLeft, MILLISECONDS)` cancel delegate future?
This is a bug in Presto. Futures returned in `DynamicFilter` are not resilient against cancellation. Could you fix it?
We probably should wrap these in `MoreFutures#unmodifiableFuture(java.util.concurrent.CompletableFuture<V>, boolean)` in Presto main

Reply:
`orTimeout` completes the underlying future exceptionally
E.g.:
```
CompletableFuture future = dynamicFilter.isBlocked();
assertFalse(future.isDone());
future.orTimeout(1, TimeUnit.MILLISECONDS);
Thread.sleep(1000);
assertFalse(future.isCancelled());
assertTrue(future.isCompletedExceptionally());
assertTrue(future.isDone());
assertFalse(dynamicFilter.isBlocked().isDone());
```
If we outright cancel the future returned by `dynamicFilter.isBlocked()` then the cancellation propagates into DynamicFilterService and prevents setting of `lazyDynamicFilters` in `addDynamicFilters`.
```
CompletableFuture future = dynamicFilter.isBlocked();
assertFalse(future.isDone());
future.cancel(false);
assertTrue(future.isCancelled());
assertTrue(dynamicFilter.isBlocked().isDone());
```
I'll make the change to warp the future in unmodifiableFuture

Reply:
Added a commit for wrapping in unmodifiableFuture